src/xtgeo/grid3d/_egrid.py

```diff
@@ -579,57 +645,68 @@ def to_file(self, filelike, file_format: Format = Format.UNFORMATTED):
+
+    @mapaxes.setter
+    def mapaxes(self, value):
+        self.egrid_head.mapaxes = value
+
     @property
-    def dimensions(self):
+    def dimensions(self) -> Tuple[int, int, int]:
         return self.global_grid.dimensions
 
+    @property
+    def map_axis_units(self) -> Units:
+        return self.egrid_head.mapunits
+
+    @map_axis_units.setter
+    def map_axis_units(self, value):
+        self.egrid_head.mapunits = value
+
+    @property
+    def grid_units(self) -> Units:
+        return self.egrid_head.gridunit.unit
+
+    @grid_units.setter
+    def grid_units(self, value):
+        self.egrid_head.gridunit.unit = value
+
     @classmethod
     def from_xtgeo_grid(cls, xtgeo_grid):
-        xtgeo_grid._xtgformat2()
-        global_grid = GlobalGrid.from_xtgeo_grid(xtgeo_grid)
-        global_grid.coord = global_grid.coord.astype(np.float32)
-        global_grid.zcorn = global_grid.zcorn.astype(np.float32)
-        rock_model = RockModel.SINGLE_PERMEABILITY_POROSITY
+        default_grid = super().from_xtgeo_grid(xtgeo_grid)
+
+        default_grid.global_grid.coord = default_grid.global_grid.coord.astype(
+            np.float32
+        )
+        default_grid.global_grid.zcorn = default_grid.global_grid.zcorn.astype(
+            np.float32
+        )
         if xtgeo_grid._dualporo:
-            rock_model = RockModel.DUAL_POROSITY
+            default_grid.rock_model = RockModel.DUAL_POROSITY
         if xtgeo_grid._dualperm:
-            rock_model = RockModel.DUAL_PERMEABILITY
-        return EGrid(
-            EGridHead(
-                Filehead(
-                    3,
-                    2007,
-                    3,
-                    TypeOfGrid.CORNER_POINT,
-                    rock_model,
-                    GridFormat.IRREGULAR_CORNER_POINT,
-                )
-            ),
-            global_grid,
-            [],
-            [],
-        )
-
-    def xtgeo_coord(self) -> np.ndarray:
-        previous_mapaxes = self.global_grid.mapaxes
-        self.global_grid.mapaxes = self.egrid_head.mapaxes
-        result = self.global_grid.xtgeo_coord()
-        self.global_grid.mapaxes = previous_mapaxes
-        return result
-
-    def xtgeo_actnum(self) -> np.ndarray:
-        previous_mapaxes = self.global_grid.mapaxes
-        self.global_grid.mapaxes = self.egrid_head.mapaxes
-        result = self.global_grid.xtgeo_actnum()
-        self.global_grid.mapaxes = previous_mapaxes
-        return result
-
-    def xtgeo_zcorn(self) -> np.ndarray:
-        previous_mapaxes = self.global_grid.mapaxes
-        self.global_grid.mapaxes = self.egrid_head.mapaxes
-        result = self.global_grid.xtgeo_zcorn()
-        self.global_grid.mapaxes = previous_mapaxes
-        return result
+            default_grid.rock_model = RockModel.DUAL_PERMEABILITY
+        else:
+            warnings.warn(
+                "Unitless xtgeo grid converted to egrid. Assuming meters as unit."
+            )
```

Comment:
Suspecting this else is incorrectly placed.. Looks like the previous commit had `GridUnits` added, and that is the check it belongs to.

Reply:
I think this warning is in the correct commit as it pertains to conversion of units.

Reply:
Unless I'm getting some weird results here..

The commit with unit conversion has the following:

```
 gridunits = GridUnit()
 if xtgeo_grid.units is not None:
     gridunits.units = xtgeo_grid.units
 else:
     warnings.warn(
         "Unitless xtgeo grid converted to egrid. Assuming meters as unit."
     )

```
Which I agree to!

However the final result after this PR is:

```
 if xtgeo_grid._dualperm:
         default_grid.rock_model = RockModel.DUAL_PERMEABILITY
 else:
     warnings.warn(
         "Unitless xtgeo grid converted to egrid. Assuming meters as unit."
     )
```

Right?

Reply:
I see that the problem is the `else` which I only now noticed. Fixing!

Reply:
Fixed now.